### PR TITLE
Lay the first stone of FakeGenerator

### PR DESCRIPTION
**Pull request by dmlyapin**
Wait for continuation in the next episodes...

### CODE REVIEW

File: src/fakeWriter.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+  }
+  const formatted = await format(lines.join('\n'));
+  fs.writeFileSync(fileName, formatted);
+}
+
+// private
+
+function isFakeRequired(definition: any): boolean {
+  return Boolean(definition.properties?.id);
```
Comment (dmlyapin) [inline src/fakeWriter.ts:24]: If a model has `id`, let's build a `fake` for it.

File: src/fakeWriter.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+function writeFake(lines: string[], name: string, definition: any): void {
+  lines.push(`// fake${name}`);
+  lines.push(`// ${JSON.stringify(definition)}`);
```
Comment (dmlyapin) [inline src/fakeWriter.ts:29]: just comments as a first step

File: src/tests/__snapshots__/generator.test.ts.snap [hunk 3]
```diff
@@ -513,6 +513,15 @@ export enum ProcessDefinitionKey {
 "
 `;
 
+exports[`Fakes 1`] = `
```
Comment (dmlyapin) [inline src/tests/__snapshots__/generator.test.ts.snap:516]: generated `fakes.ts`

File: src/tests/generator.test.ts [hunk 4]
```diff
@@ -159,6 +159,16 @@ it('Typed Schemas', async () => {
...
 });
 
+it('Fakes', async () => {
+  const models = `
+    WithoutId { name: s }
+    User { id: i, email: s }
+    Company { id: i, name: s }
+  `;
+  const { fakesSource } = await generate({ endpoints: '', models });
+  expect(fakesSource).toMatchSnapshot();
```
Comment (dmlyapin) [inline src/tests/generator.test.ts:169]: https://github.com/OsomePteLtd/openapi-ts-sdk/pull/67/files#r1561145492

File: src/fakeWriter.ts (hunk 1 above)
Comment (Rendalf) [reply src/fakeWriter.ts:24]: `definition.properties?.id` -> `definition?.properties?.id` since it's any?

**Review by Rendalf — APPROVED** [review]

File: src/fakeWriter.ts (hunk 1 above)
Comment (dmlyapin) [reply src/fakeWriter.ts:24]: definition is ok, 🦷 

and without "?" for `properties` tests are failing 🙂

**Review by divineforest — APPROVED** [review]
nice! 🎉

**Pull request merged**